contracts/bonding/BondingManager.sol

```diff
@@ -613,15 +654,14 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
      * @param _newPosPrev Address of previous transcoder in pool if the delegate is in the pool
      * @param _newPosNext Address of next transcoder in pool if the delegate is in the pool
      */
-    function rebondWithHint(
+    function _rebondWithHint(
+        address _delegator,
         uint256 _unbondingLockId,
         address _newPosPrev,
         address _newPosNext
-    ) public whenSystemNotPaused currentRoundInitialized autoClaimEarnings {
-        require(delegatorStatus(msg.sender) != DelegatorStatus.Unbonded, "caller must be bonded");
-
+    ) internal {
         // Process rebond using unbonding lock
-        processRebond(msg.sender, _unbondingLockId, _newPosPrev, _newPosNext);
+        processRebond(_delegator, _unbondingLockId, _newPosPrev, _newPosNext);
```

Comment:
WDYT about removing this internal function `_rebondWithHint` entirely and calling `processRebond` inside `rebondWithHint` and `transferBond` directly as all it does is call `processBond` anyway